From 9729239306edd6c0d122b0222626988f44d9b1cf Mon Sep 17 00:00:00 2001
From: Sven Eckelmann <sven@narfation.org>
Date: Thu, 22 Sep 2016 23:41:20 +0200
Subject: [PATCH] alfred: vis: Fix netlink function fallback

The netlink functions return -EOPNOTSUPP when the kernel doesn't support
the netlink command. The caller can then fall back to debugfs. So check for
-EOPNOTSUPP and not EOPNOTSUPP.

Fixes: bca55a86fecd ("alfred: vis: Add support for netlink")
Signed-off-by: Sven Eckelmann <sven@narfation.org>
---
 vis/vis.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/vis/vis.c b/vis/vis.c
index b865508..677f8aa 100644
--- a/vis/vis.c
+++ b/vis/vis.c
@@ -361,7 +361,7 @@ static int parse_transtable_local(struct globals *globals)
 	int ret;
 
 	ret = parse_transtable_local_netlink(globals);
-	if (ret != EOPNOTSUPP)
+	if (ret != -EOPNOTSUPP)
 		return ret;
 
 	return parse_transtable_local_debugfs(globals);
@@ -602,7 +602,7 @@ static int parse_orig_list(struct globals *globals)
 	int ret;
 
 	ret = parse_orig_list_netlink(globals);
-	if (ret != EOPNOTSUPP)
+	if (ret != -EOPNOTSUPP)
 		return ret;
 
 	return parse_orig_list_debugfs(globals);
-- 
2.9.3

